### PR TITLE
CMC-354: add springboot security OIDC

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
[CMC-354](https://tools.hmcts.net/jira/browse)

### Change description ###
This PR implements the springboot security and uses the OIDC onfigurations to fetch and verify the user details.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: charts/ucmc-service/Chart.yaml [hunk 1]
```diff
@@ -2,6 +2,6 @@ apiVersion: v2
 description: A Helm chart for ucmc-service App
 name: ucmc-service
 home: https://github.com/hmcts/ucmc-service
-version: 0.1.0
+version: 0.2.0
```
Comment (dharmendrak) [inline charts/ucmc-service/Chart.yaml:5]: shouldn't it be 0.1.1

Comment (sabahirfan) [reply charts/ucmc-service/Chart.yaml:5]: If we are **not** using semantic versioning here then yes.

**Review by dharmendrak — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/ucmc/config/SecurityConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import uk.gov.hmcts.reform.ucmc.security.JwtGrantedAuthoritiesConverter;
+
+import static org.springframework.security.config.http.SessionCreationPolicy.STATELESS;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+    private static final String[] AUTHORITIES = {
+        "caseworker-cmc",
```
Comment (ckalista) [inline src/main/java/uk/gov/hmcts/reform/ucmc/config/SecurityConfiguration.java:29]: Roles should be `caseworker-civil` & `caseworker-civil-solicitor`.

**Review by ckalista — DISMISSED** [review]
LGTM - just check the roles are correct.

File: src/main/java/uk/gov/hmcts/reform/ucmc/config/SecurityConfiguration.java (hunk 2 above)
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/reform/ucmc/config/SecurityConfiguration.java:29]: That will change after we merge this PR: https://github.com/hmcts/civil-unspecified-docker/pull/2

**Review by ckalista — APPROVED** [review]

**Review by dharmendrak — APPROVED** [review]

**Pull request merged**